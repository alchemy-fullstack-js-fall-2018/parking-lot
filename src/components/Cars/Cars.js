import React from 'react';
import PropTypes from 'prop-types';
import Car from '../Car/Car';

const Cars = ({ cars, summary }) => {

  const carItems = cars.map(car => {
    return <Car key={car.plate} car={car} />;
  });

  return (
    <div>
      <h3>{summary}</h3>
      <ul>
        {carItems}
      </ul>
    </div>
  );
};

Cars.propTypes = {
  cars: PropTypes.array.isRequired,
  summary: PropTypes.string
};

export default Cars;
