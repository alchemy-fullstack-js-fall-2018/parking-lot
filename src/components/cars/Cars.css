table {
  border: 1px solid black;
  text-align: center;
  width: 600px;
  font-size: 20px;
}
