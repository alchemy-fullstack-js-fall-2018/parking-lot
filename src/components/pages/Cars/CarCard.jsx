import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../../routes';
import styled from 'styled-components';

const StyledDiv =  styled.div`
  margin: 5px;
  border: 1px solid black;
`;

const CarCard = ({ plate, make, model, id }) => {
  return (
    <StyledDiv>
      <p>plate: {plate}</p>
      <p>make: {make}</p>
      <p>model: {model}</p>
      <Link
        to={ROUTES.CAR_DETAIL.linkTo(id)}
      >See more information
      </Link>
    </StyledDiv>
  );
};

CarCard.propTypes = {
  plate: PropTypes.string.isRequired,
  make: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default CarCard;
