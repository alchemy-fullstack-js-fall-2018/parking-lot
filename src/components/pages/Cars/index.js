export { default } from './CarList/CarList';
