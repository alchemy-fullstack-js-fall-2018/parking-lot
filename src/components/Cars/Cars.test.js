import React from 'react';
import { shallow } from 'enzyme';
import Cars from './Cars';

jest.mock('../../routes');

describe ('tests for Cars component', () => {
  const props = {
    summary: 'Test Cars',
    cars: [{
      plate: 'ZZZ000',
      make: 'Plymouth',
      model:  'Edsel',
      color: 'pink',
      lotUseCount: 2,
      isInLot: true
    },
    {
      plate: 'YYY111',
      make: 'Plymouth',
      model:  'Acclaim',
      color: 'red',
      lotUseCount: 3,
      isInLot: false
    }]
  };

  it('renders a Cars component', () => {
    const wrapper = shallow(<Cars cars={props.cars} summary={props.summary}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
