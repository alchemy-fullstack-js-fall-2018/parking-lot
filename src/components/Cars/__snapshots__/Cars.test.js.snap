// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Cars component renders a Cars component 1`] = `
<div>
  <h3>
    Test Cars
  </h3>
  <ul>
    <Car
      car={
        Object {
          "color": "pink",
          "isInLot": true,
          "lotUseCount": 2,
          "make": "Plymouth",
          "model": "Edsel",
          "plate": "ZZZ000",
        }
      }
      key="ZZZ000"
    />
    <Car
      car={
        Object {
          "color": "red",
          "isInLot": false,
          "lotUseCount": 3,
          "make": "Plymouth",
          "model": "Acclaim",
          "plate": "YYY111",
        }
      }
      key="YYY111"
    />
  </ul>
</div>
`;
